src/storage/src/hummock/sstable/builder.rs

```diff
@@ -158,9 +153,14 @@ impl SstableBuilder {
     /// ```
     pub fn finish(mut self) -> (u64, Bytes, SstableMeta, Vec<u32>) {
         let smallest_key = self.block_metas[0].smallest_key.clone();
-        let largest_key = self.last_full_key.to_vec();
+        let largest_key = if self.block_builder.is_empty() {
+            self.last_full_key.clone()
+        } else {
+            self.block_builder.get_last_key().to_vec()
+        };
         self.build_block();
```

Comment:
Can we call `build_block` first and then we don't need to check whether `self.block_builder.is_empty()`?

Reply:
No. We can not build block at first because we can not get the last key from bloc.